sdk/automation/Azure.ResourceManager.Automation/src/Generated/ModuleResource.cs

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager;
+
+namespace Azure.ResourceManager.Automation
+{
+    /// <summary> TODO. </summary>
+    public abstract partial class ModuleResource : ArmResource
+    {
+        internal static ModuleResource GetResource(ArmClient client, ModuleData data)
+        {
+            if (IsAutomationAccountPython2PackageResource(data.Id))
+            {
+                return new AutomationAccountPython2PackageResource(client, data);
+            }
+            if (IsAutomationAccountModuleResource(data.Id))
+            {
+                return new AutomationAccountModuleResource(client, data);
+            }
+            // TODO -- should we throw or return an UnknownResource?
+            throw new InvalidOperationException();
+        }
+
+        internal static bool IsAutomationAccountPython2PackageResource(ResourceIdentifier id)
+        {
+            // checking the resource type
+            if (id.ResourceType != AutomationAccountPython2PackageResource.ResourceType)
+            {
+                return false;
+            }
+            // checking the resource scope
+            if (id.Parent.Parent.ResourceType != "Microsoft.Resources/resourceGroups")
```

Comment:
Does it make sense to use `ResourceGroupResource.ResourceType` instead of a raw string?

Reply:
Will do